nexus_motion_planner/src/motion_plan_cache.hpp

```diff
@@ -81,7 +78,7 @@ namespace motion_planner {
  *
  * Motion plans may be looked up with some tolerance at call time.
  */
-class MotionPlanCache
+class MotionPlanCache : public std::enable_shared_from_this<MotionPlanCache>
```

Comment:
Since this has `enable_shared_from_this`, I'm assuming that it must be created as a `shared_ptr`. If so, make the constructor private and have a public static "make" function to create an instance. This prevents accidentally creating a non `shared_ptr` instance which result in undefined behaviors.

---

On second look, I don't see any `*_from_this` used? Do we need this inheritance?

Reply:
I had a brainfart and put it in when I had to make a shared_ptr of the class.

Fixed.
bf53bf7a4906794ff48fd6a2d05ecd39d4191fa3